store/tikv/gcworker/gc_worker.go

```diff
@@ -642,6 +643,26 @@ func (w *GCWorker) resolveLocks(ctx context.Context, safePoint uint64) error {
 	return nil
 }
 
+func (w *GCWorker) syncSafePointWithPd(ctx context.Context, safePoint uint64) error {
+	var newSafePoint uint64
+	var err error
+	// Try to communicate with PD at most 3 times.
+	for i := 0; i < 3; i++ {
+		newSafePoint, err = w.pdClient.UpdateGCSafePoint(ctx, safePoint)
+		if err == nil {
+			break
+		}
+	}
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if newSafePoint != safePoint {
+		log.Warnf("[gc worker] pd rejected our safe point %v, because pd has greater safe point %v", safePoint, newSafePoint)
```

Comment:
Is it possible that pd reject the update safepoint request and pd's safepoint is less than tidb's safepoint?